### PR TITLE
fix(ServiceExport): Reject Conflicting Alias Names for ServiceExport object (#315)

### PR DESCRIPTION
**Pull request by soharab-ic**
# Description
 The validating webhook will validate the ServiceExport object and admission of ServiceExport objects with conflicting alias names will be rejected.

Fixes   315

## Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR requires documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have ran `go fmt`
* [ ] I have updated the helm chart as required by this PR.
* [x] I have performed a self-review of my own code.
* [x] I have commented my code, particularly in hard-to-understand areas.
* [ ] I have tested it for all user roles.
* [x] I have added all the required unit test cases.
* [ ] I have verified the E2E test cases with new code changes.
* [ ] I have added all the required E2E test cases.

## Does this PR introduce a breaking change?

```release-note

```

### CODE REVIEW

File: pkg/webhook/pod/webhook.go [hunk 1]
```diff
@@ -261,6 +281,33 @@ func MutateDaemonSet(ds *appsv1.DaemonSet, sliceName string) *appsv1.DaemonSet {
...
+	serviceExportList, err := wh.SliceInfoClient.GetAllServiceExports(context.Background(), wh.Client, svcex.Spec.Slice)
+	if err != nil {
+		return false, "", err
+	}
+
+	newAliases := svcex.Spec.Aliases
+
+	for _, serviceExport := range serviceExportList.Items {
+		// In case we are updating an existing ServiceExport resource
+		if svcex.ObjectMeta.Name == serviceExport.ObjectMeta.Name &&
```
Comment (YachikaRalhan) [inline pkg/webhook/pod/webhook.go:293]: Why are you skipping the existing service export updation validation?

Comment (soharab-ic) [reply pkg/webhook/pod/webhook.go:293]: If we are updating a serviceexport object svcex-1:
1. serviceExportList will contain svcex-1.
2. Comparing the aliases of updated svcex-1 and existing svcex-1 may give us conflict.
3. Even if we are updating the aliases, it is safe to compare with aliases of other serviceexport objects not the same seviceexport object.

**Comment by rahulii** [comment]
Looks good overall, please test it once !
Also please add unit test cases!

**Comment by soharab-ic** [comment]
The issue 338 impacts this PR. Will complete the testing once it is resolved.

**Comment by soharab-ic** [comment]
> Looks good overall, please test it once ! Also please add unit test cases!

Already added unit test cases.
https://github.com/infracloudio/kubeslice-worker-operator/pull/2/files#diff-45a421353c537e3010114d2de28717519d2e86df663d7130d7cc64bca4f8e48cR158

**Review by rahulii — APPROVED** [review]

**Pull request merged**